modules/nf-core/tcoffee/consensus/tests/main.nf.test

```diff
@@ -0,0 +1,184 @@
+            }
+            run("TCOFFEE_ALIGN") {
+                script "../../../tcoffee/align/main.nf"
+                process {
+                    """
+                    input[0] = [ [ id:'test' ],
+                                 file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/setoxin-ref.fa", checkIfExists: true)                                 
+                               ]
+                    input[1] = [[:],[]]
+                    input[2] = [[:],[],[]]
+                    input[3] = false
+
+                    """
+                }
+            }          
+        }
+        when {
+            process {
+                """
+                msas = FAMSA_ALIGN.out.alignment.mix(TCOFFEE_ALIGN.out.alignment).groupTuple()
+                input[0] = msas
+                input[1] = [[:],[]]
+                input[2] = false
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out.alignment).match("alignment")},
+                { assert snapshot(process.out.versions).match("versions_uncomp") }
+            )
+        }
+    }
+
+     test("consensus - tree - compressed") {
+
+        config "./sequence.config"
+        setup {
+            run("FAMSA_GUIDETREE") {
+                script "../../../famsa/guidetree/main.nf"
+                process {
+                    """
+                    input[0] = [ [ id:'test' ],
+                                 file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/setoxin-ref.fa", checkIfExists: true)                                 
+                               ]
+                    """
+                }
+            }
+            run("FAMSA_ALIGN") {
+                script "../../../famsa/align/main.nf"
+                process {
+                    """
+                    input[0] = [ [ id:'test' ],
+                                 file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/setoxin-ref.fa", checkIfExists: true)                                 
+                               ]
+                    input[1] = [[:],[]]
+                    input[2] = false
+
+                    """
+                }
+            }
+            run("TCOFFEE_ALIGN") {
+                script "../../../tcoffee/align/main.nf"
+                process {
+                    """
+                    input[0] = [ [ id:'test' ],
+                                 file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/setoxin-ref.fa", checkIfExists: true)                                 
+                               ]
+                    input[1] = [[:],[]]
+                    input[2] = [[:],[],[]]
+                    input[3] = false
+
+                    """
+                }
```

Comment:
The setup can be moved to the beginning of the test file and then re-used for multiple tests :)

Reply:
Uh nice!